components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/test/java/org/wso2/carbon/identity/application/authentication/framework/handler/sequence/impl/DefaultStepBasedSequenceHandlerTest.java

```diff
@@ -323,4 +449,341 @@ public void testResetAuthenticationContext() throws Exception {
+                                          String expectedUserStoreToBeProvisioned) throws Exception {
+
+        final String subjectIdentifier = "subjectID";
+        final List<String> mappedRoles = Collections.emptyList();
+        final String TENANT_DOMAIN = "foo.com";
+
+        context = getMockedContextForJitProvisioning(provisioningUserStoreId, provisioningUserStoreClaimUri, TENANT_DOMAIN);
+
+        ArgumentCaptor<String> captor = ArgumentCaptor.forClass(String.class);
+
+        // Mock the provisioning handler
+        ProvisioningHandler provisioningHandler = mock(ProvisioningHandler.class);
+        doNothing().when(provisioningHandler).handle(anyList(), anyString(), anyMap(), captor.capture(), anyString());
+
+        // Mock framework util to returned mocked provisoning handler
+        returnMockedProvisoningHandlerFromFramework(provisioningHandler);
+        mockHandlerThreadLocalProvisioningServiceProvider();
+
+        stepBasedSequenceHandler.handleJitProvisioning(subjectIdentifier, context, mappedRoles, externalAttributeValues);
+        verify(provisioningHandler).handle(anyList(), anyString(), anyMap(), captor.capture(), anyString());
+
+        // check whether the user is provisioned to correct user store
+        assertEquals(captor.getValue(), expectedUserStoreToBeProvisioned);
+        assertNotNull(threadLocalProvisioningSp);
+        assertTrue(threadLocalProvisioningSp.isJustInTimeProvisioning());
+        assertEquals(threadLocalProvisioningSp.getTenantDomain(), TENANT_DOMAIN);
+        assertEquals(threadLocalProvisioningSp.getClaimDialect(), ApplicationConstants.LOCAL_IDP_DEFAULT_CLAIM_DIALECT);
+    }
+
+    private void mockHandlerThreadLocalProvisioningServiceProvider() throws Exception {
+
+        mockStatic(IdentityApplicationManagementUtil.class);
+
+        doAnswer(new Answer<Object>() {
+            public Object answer(InvocationOnMock invocation) throws Throwable {
+                threadLocalProvisioningSp = (ThreadLocalProvisioningServiceProvider) invocation.getArguments()[0];
+                return null;
+            }
+        }).when(IdentityApplicationManagementUtil.class, "setThreadLocalProvisioningServiceProvider",
+                any(ThreadLocalProvisioningServiceProvider.class));
+    }
+
+    private AuthenticationContext getMockedContextForJitProvisioning(String provisioningUserStoreId,
+                                                                     String provisioningUserStoreClaimUri,
+                                                                     String tenantDomain) {
+
+        ExternalIdPConfig externalIdPConfig = spy(new ExternalIdPConfig());
+        when(externalIdPConfig.getProvisioningUserStoreId()).thenReturn(provisioningUserStoreId);
+        when(externalIdPConfig.getProvisioningUserStoreClaimURI()).thenReturn(provisioningUserStoreClaimUri);
+
+        ApplicationConfig applicationConfig = new ApplicationConfig(mock(ServiceProvider.class));
```

Comment:
Can't we use `new ServiceProvider()` instead of mocking?

Reply:
+1